File: SDKs/amazonaws.com/glue/2017-03-31/java/lib/src/main/java/openapisdk/models/shared/PrincipalPermissions.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * PrincipalPermissions
 * Permissions granted to a principal.
**/
public class PrincipalPermissions {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("Permissions")
    public PermissionEnum[] permissions;
    public PrincipalPermissions withPermissions(PermissionEnum[] permissions) {
        this.permissions = permissions;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("Principal")
    public DataLakePrincipal principal;
    public PrincipalPermissions withPrincipal(DataLakePrincipal principal) {
        this.principal = principal;
        return this;
    }
}

File: SDKs/amazonaws.com/glue/2017-03-31/python/src/sdk/models/shared/principalpermissions.py
import dataclasses
from typing import Optional
from enum import Enum
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import permission_enum as shared_permission_enum
from ..shared import datalakeprincipal as shared_datalakeprincipal


@dataclass_json
@dataclasses.dataclass
class PrincipalPermissions:
    r"""PrincipalPermissions
    Permissions granted to a principal.
    """
    
    permissions: Optional[list[shared_permission_enum.PermissionEnum]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('Permissions') }})
    principal: Optional[shared_datalakeprincipal.DataLakePrincipal] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('Principal') }})
    
